### ISSUE TITLE
[Phase 2] Survival Model Implementation

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/predictive-maintenance-survival#3]
## Summary

Implement survival analysis models (Random Survival Forest and Cox Proportional Hazards) with proper handling of censored observations and custom cross-validation.

**Parent Issue:** #1
**Depends on:** #2 (Phase 1: Data Pipeline)

## Objectives

- Train Random Survival Forest for time-to-event prediction
- Train Cox Proportional Hazards model for interpretable hazard ratios
- Implement censoring-aware cross-validation
- Evaluate with C-index and integrated Brier score

## Key Concepts

**Right Censoring**: Test engines haven't failed yet - we know survival time is *at least* this long but not the true failure time. This must be handled properly in training.

**Survival Function S(t)**: Probability of surviving past time t. Used to compute RUL.

## Files to Create

| File | Description |
|------|-------------|
| `src/models/survival_forest.py` | RSF wrapper with consistent API |
| `src/models/cox_model.py` | Cox PH with feature importance |
| `src/evaluation/metrics.py` | C-index, integrated Brier score |
| `src/evaluation/cross_validation.py` | Censoring-stratified CV |
| `tests/test_survival_models.py` | Model tests |

### Survival Forest
- [ ] Wrap `sksurv.ensemble.RandomSurvivalForest`
- [ ] Implement `fit(X, duration, event)` interface
- [ ] Implement `predict_survival_function()` returning curves
- [ ] Implement `predict_median_survival_time()`
- [ ] Hyperparameter tuning with CV

### Cox Model
- [ ] Wrap `lifelines.CoxPHFitter`
- [ ] Extract hazard ratios for feature importance
- [ ] Check proportional hazards assumption
- [ ] Handle multicollinearity (VIF check)

### Evaluation
- [ ] C-index (concordance) implementation
- [ ] Integrated Brier score for calibration
- [ ] Survival curve visualization

### Cross-Validation
- [ ] Stratify by censoring status
- [ ] Stratify by duration quantiles
- [ ] Ensure no engine ID leakage

## Code Skeleton

📄 **File:** `src/models/survival_forest.py`

```python
"""Random Survival Forest wrapper."""

import numpy as np
from sksurv.ensemble import RandomSurvivalForest
from sksurv.util import Surv


class SurvivalForestModel:
    """Wrapper for Random Survival Forest with consistent API."""

    def __init__(
        self,
        n_estimators: int = 100,
        max_depth: int = None,
        min_samples_split: int = 10,
        random_state: int = 42
    ):
        self.model = RandomSurvivalForest(
            n_estimators=n_estimators,
            max_depth=max_depth,
            min_samples_split=min_samples_split,
            random_state=random_state,
            n_jobs=-1
        )
        self._is_fitted = False

    def fit(self, X: np.ndarray, duration: np.ndarray, event: np.ndarray):
        """Fit the survival forest.

        Args:
            X: Feature matrix (n_samples, n_features)
            duration: Time to event or censoring (non-negative)
            event: Boolean array (True = event occurred, False = censored)
        """
        # Create structured array for sksurv
        y = Surv.from_arrays(event.astype(bool), duration)
        self.model.fit(X, y)
        self._is_fitted = True
        return self

    def predict_survival_function(self, X: np.ndarray):
        """Predict survival function S(t) for each sample.

        Returns:
            List of StepFunction objects (one per sample)
        """
        if not self._is_fitted:
            raise RuntimeError("Model must be fitted first")
        return self.model.predict_survival_function(X)

    def predict_median_survival_time(self, X: np.ndarray) -> np.ndarray:
        """Predict median survival time (when S(t) = 0.5).

        Returns:
            Array of median survival times
        """
        surv_funcs = self.predict_survival_function(X)
        medians = []

        for sf in surv_funcs:
            # Find where survival probability crosses 0.5
            times = sf.x
            probs = sf.y

            # Find crossing point
            idx = np.searchsorted(-probs, -0.5)
            if idx >= len(times):
                medians.append(times[-1])  # Never crosses 0.5
            else:
                medians.append(times[idx])

        return np.array(medians)

    def score(self, X: np.ndarray, duration: np.ndarray, event: np.ndarray) -> float:
        """Compute C-index on given data."""
        y = Surv.from_arrays(event.astype(bool), duration)
        return self.model.score(X, y)
```

📄 **File:** `src/evaluation/cross_validation.py`

```python
"""Censoring-aware cross-validation for survival analysis."""

import numpy as np
from sklearn.model_selection import StratifiedKFold


class CensoringAwareCV:
    """Cross-validation that stratifies by censoring status and duration."""

    def __init__(self, n_splits: int = 5, random_state: int = 42):
        self.n_splits = n_splits
        self.random_state = random_state

    def split(self, X: np.ndarray, duration: np.ndarray, event: np.ndarray):
        """Generate train/test indices with censoring-aware stratification.

        Stratifies by: event status + duration quartile (8 groups total)
        """
        # Create stratification labels
        duration_quartiles = np.digitize(
            duration,
            np.percentile(duration, [25, 50, 75])
        )

        # Combine event status and quartile: 0-3 for censored, 4-7 for events
        strat_labels = event.astype(int) * 4 + duration_quartiles

        skf = StratifiedKFold(
            n_splits=self.n_splits,
            shuffle=True,
            random_state=self.random_state
        )

        for train_idx, test_idx in skf.split(X, strat_labels):
            yield train_idx, test_idx
```

## Definition of Done

- [ ] Survival Forest achieves C-index > 0.70 on FD001 test set
- [ ] Cox model produces non-null hazard ratios for all features
- [ ] Cross-validation uses stratified censoring splits
- [ ] Models serialize/deserialize correctly with joblib
- [ ] All tests pass

## Risk Assessment

| Risk | Impact | Mitigation |
|------|--------|------------|
| sksurv version incompatibility | 🔴 | Pin exact version in requirements |
| Cox PH assumption violation | 🟡 | Run Schoenfeld residual test |
| Multicollinearity in Cox | 🟡 | VIF check, drop correlated features |
| C-index below threshold | 🔴 | More feature engineering, hyperparameter tuning |
